tests/test_openupgradelib.py

```diff
@@ -31,16 +38,213 @@ def import_mock(name, *args):
 
     def setUp(self):
-        pass
+        self.cr = PsycopgCrMock()
+        # Declare mock databases entries
+        self.tables = {
+            "ir_module_module": ["name"],
+            "ir_model_data": ["name", "module", "model"],
+            "ir_module_module_dependency": ["name"],
+            "ir_translation": ["module"],
+        }
+        # Declare test data in mock database
+        self.data = {
+            "ir_module_module": [
+                {"name": "old_module_name"},
+                {"name": "other_module_name"},
+            ],
+            "ir_model_data": [
+                {
+                    "module": "base",
+                    "name": "module_old_module_name",
+                    "model": "ir.module.module",
+                },
+                {
+                    "module": "base",
+                    "name": "module_other_module_name",
+                    "model": "ir.module.module",
+                },
+                {
+                    "module": "old_module_name",
+                    "name": "some_xid1",
+                    "model": "a_model",
+                },
+                {
+                    "module": "old_module_name",
+                    "name": "some_xid2",
+                    "model": "a_model",
+                },
+                {
+                    "module": "other_module_name",
+                    "name": "some_xid",
+                    "model": "a_model",
+                },
+            ],
+            "ir_module_module_dependency": [
+                {"name": "old_module_name"},
+                {"name": "other_module_name"},
+            ],
+            "ir_translation": [
+                {"module": "old_module_name"},
+                {"module": "other_module_name"},
+            ],
+        }
+        self._create_mock_database_tables(self.tables)
+        self._insert_mock_database_entries(self.data)
+
+    def _create_mock_database_tables(self, table_spec):
+        for table, columns in table_spec.items():
+            self.cr.execute("CREATE TABLE %(table)s(%(columns)s)" % {
+                "table": table,
+                "columns": ", ".join(columns),
+            })
+
+    def _insert_mock_database_entries(self, data_spec):
+        for table, rows in data_spec.items():
+            for row in rows:
+                columns, entries = zip(*row.items())
+                self.cr.sqlite_cr.execute("""\
+INSERT INTO %(table)s (%(columns)s)
+VALUES (%(entries)s)
+""" % {
+                    "table": table,
+                    "columns": ", ".join(columns),
+                    "entries": ", ".join(["?"] * len(entries))
+                }, entries)
+
+    def test_update_module_names(self):
+        namespec = [("old_module_name", "new_module_name")]
+        CompSpec = namedtuple(
+            "CompSpec",
+            ["table", "module_field", "where", "name_modifier", "size_comp"]
+        )
+
+        def assert_entry_counts(compspecs, pre=False):
```

Comment:
Take a look at it. It is very specific to this test and is not reusable in its current form.